sklearn/cluster/k_means_.py

```diff
@@ -441,10 +441,19 @@ def _kmeans_single(X, n_clusters, x_squared_norms, max_iter=300,
             best_centers = centers.copy()
             best_inertia = inertia
 
-        if squared_norm(centers_old - centers) <= tol:
+        shift = squared_norm(centers_old - centers)
+        if shift <= tol:
             if verbose:
                 print("Converged at iteration %d" % i)
+
             break
+
+    if shift > 0:
```

Comment:
Noticed something here – if ``max_iter`` is set to zero, then ``shift`` will be undefined. We should probably add a check on max_iter (probably within ``KMeans``) and raise a useful error if it is less than 1.